quinn-udp/src/cmsg/unix.rs

```diff
@@ -43,7 +43,15 @@ impl MsgHdr for crate::imp::msghdr_x {
 
     fn cmsg_nxt_hdr(&self, cmsg: &Self::ControlMessage) -> *mut Self::ControlMessage {
         let selfp = self as *const _ as *mut libc::msghdr;
-        unsafe { libc::CMSG_NXTHDR(selfp, cmsg) }
+        let next = unsafe { libc::CMSG_NXTHDR(selfp, cmsg) };
+
+        if next.is_null()
+            || unsafe { (*next).cmsg_len as usize } < std::mem::size_of::<libc::cmsghdr>()
```

Comment:
Perhaps this is better expressed as:

```rust
if next.as_mut().is_some_and(|n| n.cmesg_len < ..)
```

Alternatively, you could extract the if null return null into a separate branch.

I think whilst relying on short-circuting here is safe, it is also not very inuitive.

Reply:
Good idea. Thanks @thomaseizinger !